i18n/create_scratch_msgs.js

```diff
@@ -0,0 +1,64 @@
+const fs = require('fs');
+const path = require('path');
+const glob = require('glob');
+const assert = require('assert');
+
+// Globals
+const PATH_INPUT = path.resolve(__dirname, '../msg/json/*.json');
+const PATH_OUTPUT = path.resolve(__dirname, '../msg');
+
+let en = fs.readFileSync(path.resolve(__dirname, '../msg/json/en.json'));
+en = JSON.parse(en);
+const enKeys = Object.keys(en).sort().toString();
+
+// Check that translation is valid:
+// elt: array [key, translation] from <locale>.json
+// - messages with placeholders have the same number of placeholders
+const validatePlaceholders = function (elt) {
+    const re = /(%\d)/g;
+    const [key, translation] = elt;
+    const placeholdersCount = en[key].match(re) ? en[key].match(re).length : 0;
+    if (placeholdersCount > 0) {
+      const tMatch = translation.match(re);
+      const enMatch = en[key].match(re);
+      assert.notStrictEqual(tMatch, null, `${key} is missing a placeholder: ${translation}`);
+      assert.strictEqual(
+          tMatch.sort().toString(),
+          enMatch.sort().toString(),
+          `${key} is missing or has duplicate placeholders: ${translation}`
+      );
+    }
+};
+
+const validate = function (json, name) {
+    // this is a little stricter than we need - it would be harmless if the translation had extra keys
```

Comment:
Actually extra keys have caused real problems in the past, because later keys references do not get replaced. This is the appropriate amount of testing.